modules/nf-core/diamond/blastp/main.nf

```diff
@@ -21,14 +21,16 @@ process DIAMOND_BLASTP {
     tuple val(meta), path('*.sam')  , optional: true, emit: sam
     tuple val(meta), path('*.tsv')  , optional: true, emit: tsv
     tuple val(meta), path('*.paf')  , optional: true, emit: paf
-    path "versions.yml"               , emit: versions
+    path "versions.yml"             , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def is_compressed = fasta.name.endsWith(".gz")
```

Comment:
there are  dedicated nxf function for this: https://nextflow.io/docs/latest/script.html#getting-file-attributes

These are safer to use rather than endsWith as this assumes input is a string (rather than a path object)

Reply:
Utilized getExtension and getBaseName functions for this. Thanks!